Terms stage (Stage 6)

hiaux0 opened the pull request:
Figma: https://www.figma.com/file/hTLvqCEzki3WoG5uJbtynw/Prime-Deals?node-id=4087%3A84321

## What was done
- Add Open Deal wizard
- By following SOT ([on notion](https://www.notion.so/primedao/Registration-Wizard-74901348d4a443a6891ad16fa1c0b42d#1c48eb0b3c3f4d1d915f0ef5ba1e4963))

## TODO
- [x] other wizards
    - [x] Open Prop: Empty
    - [x] Partnered Deal: Empty
    - [x] Make an offer (DAO1 lead): prefilled
    - [x] Make an offer (DAO2 lead): prefilled
    - [x] Edit as Lead: prefilled (not listed in mocks?)
- [x] validation
- [open] Proceed with unsaved changes
  - Unsure how to design this in code
  - https://www.notion.so/primedao/Registration-Wizard-74901348d4a443a6891ad16fa1c0b42d#9bc3280aaf3141ada8114c1975863591

## Open product questions
- [x] deleting a Clause, when there is only one Clause
  - ~~Now: Disable "Delete" button~~ --> "But deleting Clause 1 “ will "reset" the text inside." https://www.notion.so/primedao/Registration-Wizard-74901348d4a443a6891ad16fa1c0b42d#26de50dce8e2475094670c6274a9b0fc
    - cd6cf4d1bad930decf3dec48491e0e0f22279632
  - https://www.notion.so/Registration-Wizard-74901348d4a443a6891ad16fa1c0b42d?d=53ffeede100842c09f24ce3f0649b2e9#265fe726483140e2852cd816a3948a5e

## Testing

#### Before

#### After

hiaux0 commented on cypress/integration/tests/deals/terms.e2e.ts:
```diff
@@ -0,0 +1,70 @@
+const UPDATED = "(updated)";
+const MAX_NUMBER_OF_CLAUSES = 10;
+
+class Terms {
+  static addClauseButton() {
+    return cy.get("[data-test='addClauseButton']");
+  }
+  static addClause() {
+    this.addClauseButton().scrollIntoView().click();
+  }
+  static getClauses() {
+    return cy.get("[data-test='termsClause']");
+  }
+}
```
FYI: "Page Object" test pattern

Here, I used the Page Object pattern.
But as you can read in the above title [0], it has some drawbacks.
I introduced it here for the short term benefit of code reduction.
In the long term I'm eyeing to transition to the so called "Screenplay pattern" [1].
(Also because the Screenplay pattern is a more "SOLID" Page Object pattern, I'd like to learn the drawbacks of Page Objects myself)

[0]: 
Stop using Page Objects and Start using App Actions
https://www.cypress.io/blog/2019/01/03/stop-using-page-objects-and-start-using-app-actions/

[1]: 
- https://www.infoq.com/articles/Beyond-Page-Objects-Test-Automation-Serenity-Screenplay/
- https://cucumber.io/blog/bdd/understanding-screenplay-(part-1)/
- https://www.notion.so/curvelabs/Screenplay-Pattern-a717ca7dbd1248989e5c4c2584c8b50b

@dkent600 @paweljanicki @giladt @IonelLupu

IonelLupu replied:
Looks interesting and cleaner from my point of view. Nothing to say much. I will need to use it to understand it better

hiaux0 commented on src/resources/elements/editingCard/editingCard.ts:
```diff
@@ -0,0 +1,24 @@
+import { bindable } from "aurelia-framework";
+import "./editingCard.scss";
+
+export class EditingCard {
```
@dkent600 @IonelLupu 

I created a separate "<editing-card>" component.
Reason:
- Ionel used that in Stage5 (we could refactor)
- Because I basically copy pasted Ionel's card logic, I felt like, I did not want to "pollute" Terms Clauses code
- --> Separate `<editing-card>` component

IonelLupu replied:
I think there is a small problem with the Save functionality. What happens if my Save button needs to do something but then, due to some login in my Save method, I don't want to switch to the "view" mode. For example, if my validation fails.

Maybe we can make the onSave method return a boolean. If the return value is true, then we switch to the "view" mode. Other suggestions are welcomed

hiaux0 commented on cypress/integration/features/components/editingCard.feature:
```diff
@@ -0,0 +1,18 @@
+Feature: Editing Card
```
I wanted to try out the concept of Component Testing here, thus
`cypress/.../components/editingCard`

IonelLupu commented on src/wizards/tokenSwapDealWizard/stages/termsStage/termsStage.html:
```diff
@@ -0,0 +1,98 @@
+          <i class="fa fa-plus"></i>
+        </pbutton>
+
+        <!-- <pform-input
```
remove comments? Looks like these are some leftovers

IonelLupu commented on src/wizards/tokenSwapDealWizard/stages/termsStage/termsStage.ts:
```diff
@@ -0,0 +1,58 @@
+    this.wizardManager = stageMeta.wizardManager;
+    this.wizardState = this.wizardService.getWizardState(this.wizardManager);
+
+    // const validationRules = ValidationRules
```
leftovers?

IonelLupu reviewed (changes requested):
Don't forget about linting 😄

dkent600 reviewed (changes requested):
I think could look a lot nicer if the text-area fit the content more closely, like is shown in the mock

![image](https://user-images.githubusercontent.com/1821666/155233499-95cff303-1005-4c29-b77c-68fb352d38de.png)

Is supposed to show the Clause number:

![image](https://user-images.githubusercontent.com/1821666/155233898-a3a24f62-0482-4bff-a80d-82b98f4bdb73.png)

hiaux0 commented:
@dkent600 

1. removed obsolete fields
2. adjusted margins according to Mocks, and added "Clause 1" to View mode
![image](https://user-images.githubusercontent.com/30693990/155323109-f58de731-1881-411f-96e2-efc0fe1a0dd2.png)

hiaux0 commented:
1. less space between content and button
![image](https://user-images.githubusercontent.com/30693990/155394004-b2770d9f-7304-486d-a941-61ea995df701.png)

2. fixed a bug where new lines, where not rendered as such in Preview mode

dkent600 commented:
Still needs text copy
